pkg/resource/deploy/deployment.go

```diff
@@ -292,6 +293,8 @@ type Deployment struct {
 	opts *Options
 	// event handlers for this deployment.
 	events Events
+	// rebase indicates whether or not the deployment should rebase the old snapshot
+	rebase bool
```

Comment:
Similarly, do we still want to call this `rebase` or should this use some `write` related term?